youtube_dl/extractor/sbs.py

```diff
@@ -46,9 +79,197 @@ class SBSIE(InfoExtractor):
+            exp = list(exp)
+            exp.append(403)
+        else:
+            exp = None
+
+        if exp:
+            if len(args) > 7:
+                args = list(args)
+                args[7] = exp
+            else:
+                kwargs['expected_status'] = exp
+                kwargs = compat_kwargs(kwargs)
+
+        ret = super(SBSIE, self)._download_webpage_handle(url, video_id, *args, **kwargs)
+        if ret is False:
+            return ret
+        webpage, urlh = ret
+
+        if urlh.getcode() == 403:
+            if urlh.headers.get('x-error-reason') == 'geo-blocked':
+                countries = ['AU']
+                if fatal:
+                    self.raise_geo_restricted(countries=countries)
+                err = GeoRestrictedError(
+                    'This Australian content is not available from your location due to geo restriction',
+                    countries=countries)
+            else:
+                err = compat_HTTPError(urlh.geturl(), 403, 'HTTP Error 403: Forbidden', urlh.headers, urlh)
+            ret = self.__handle_request_webpage_error(err, video_id, errnote, fatal)
+            if exp:
+                # caller doesn't expect 403
+                return False
+
+        return ret
+
+    def _extract_m3u8_formats(self, m3u8_url, video_id, *args, **kwargs):
+        # ext, entry_protocol, preference, m3u8_id, note, errnote, fatal,
+        # live, data, headers, query
+        entry_protocol = args[1] if len(args) > 1 else kwargs.get('entry_protocol')
+        if not entry_protocol:
+            entry_protocol = 'm3u8_native'
+            if len(args) > 1:
+                args = list(args)
+                args[1] = entry_protocol
+            else:
+                kwargs['entry_protocol'] = entry_protocol
+                kwargs = compat_kwargs(kwargs)
+
+        return super(SBSIE, self)._extract_m3u8_formats(m3u8_url, video_id, *args, **kwargs)
+
+    AUS_TV_PARENTAL_GUIDELINES = {
+        'P': 0,
+        'C': 7,
+        'G': 0,
+        'PG': 0,
+        'M': 15,
+        'MA15+': 15,
+        'AV15+': 15,
+        'R18+': 18,
+        'NC': 0,  # not classified (unofficial, used by SBS)
+    }
+    _PLAYER_API = 'https://www.sbs.com.au/api/v3'
+    _CATALOGUE_API = 'https://catalogue.pr.sbsod.com/'
+
+    def _call_api(self, video_id, path, query=None, data=None, headers=None, fatal=True):
+        return self._download_json(update_url_query(
+            self._CATALOGUE_API + path, query),
+            video_id, headers=headers, fatal=fatal) or {}
+
+    def _get_smil_url(self, video_id):
+        return update_url_query(
+            self._PLAYER_API + 'video_smil', {'id': video_id})
+
+    def _get_player_data(self, video_id, headers=None, fatal=False):
+        return self._download_json(update_url_query(
+            self._PLAYER_API + 'video_stream', {'id': video_id, 'context': 'tv'}),
+            video_id, headers=headers, fatal=fatal) or {}
```

Comment:
I believe `headers` should default to an empty dict here and in `_call_api`. Setting `headers` to None results in options like `--geo-bypass-country` breaking.